site/_posts/2021-02-19-whats-wrong-with-implicit-grant.md

```diff
@@ -0,0 +1,86 @@
+
+> The Implicit grant ("response_type=token") is omitted from this specification as per Section 2.1.2 of OAuth 2.0 Security Best Current Practices
+
+## Run away
+
+Instead of showing you how to use this grant, let's discuss why you should avoid it.
+
+{% include _image.liquid src="/assets/img/blogs/avoid-implicit-grant/Rabbitattack.jpg" alt="Just a rabbit?" class="img-fluid" figure=false %}
+
+The reason that it has been removed is that it skips an important step that allows you to secure the tokens you receive from the OAuth server. This step occurs when your application backend makes the call to the Token endpoint to retrieve the tokens.
+
+Unlike the [Authorization Code grant](/docs/v1/tech/oauth/#example-authorization-code-grant), the Implicit grant does not redirect the browser back to your application backend with an Authorization Code. Instead, it puts the access token directly on the URL as part of the redirect. These URLs look like this:
+
+`https://piedpiper.com/#token-goes-here`
+
+The token is added to the redirect URL after the `#` symbol, which means it is technically the fragment portion of the URL. What this means is that wherever the OAuth server redirects the browser to, the access token is accessible to any code running in the browser. In other words, to basically everyone. 
+
+To be more precise, the access token is accessible to any and all JavaScript that is running in the browser (including third party libraries). Since this token allows the browser to make API calls and web requests on behalf of the user, having this token be accessible to third-party code is extremely dangerous.
+
+Let's take a dummy example of a single-page web application that uses the Implicit grant:
+
+```html
+// This is dummy code for a SPA that uses the access token
+<html>
+<head>
+  <script type="text/javascript" src="/my-spa-code-1.0.0.js"></script>
+  <script type="text/javascript" src="https://some-third-party-server.com/a-library-found-online-that-looked-cool-0.42.0.js"></script>
+</head>
+<body>
+...
+</body>
+```
+
+This HTML includes two JavaScript libraries:
+
+* The code for the application itself (`my-spa-code-1.0.0.js`).
+* A library we found online that did something cool and we just pulled it in (`a-library-found-online-that-looked-cool-0.42.0.js`).
+
+Let's assume that our code is 100% secure and we don't have to worry about it. The issue here is that the library we pulled in is an unknown quantity. 
+
+It might include other libraries as well. Remember that the DOM is dynamic. Any JavaScript can load any other JavaScript library simply by updating the DOM with more `<script>` tags. Therefore, we have very little chance of ensuring that every other line of code from third-party libraries is secure. 
```

Comment:
It might include other libraries as well. Remember ~that~ the DOM is dynamic. Any JavaScript can load any other JavaScript library simply by updating the DOM with more `<script>` tags. Therefore, we have very little chance of ensuring ~that~ every ~other~ line of code from third-party libraries is secure.